app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/SelectDutiesController.scala

```diff
@@ -54,246 +55,117 @@ class SelectDutiesController @Inject() (
-                                case Some(ndrcDetails) => ndrcDetails
-                                case None              => List.empty
-                              }
-                            case None                     => List.empty
-                          }
-
-                        if (ndrcDetails.nonEmpty) {
-                          Ok(
-                            selectDutiesPage(
-                              SelectDutiesController
-                                .selectDutiesForm(makeBlankForm(ndrcDetails, fillingOutClaim.draftClaim.isMrnFlow))
-                                .fill(DutiesSelected(List.empty)),
-                              duties(fillingOutClaim)
-                            )
-                          )
-                        } else {
-                          logger.warn(
-                            s"ACC-14 response returned no NDRC details: Num of ndrc entries: ${ndrcDetails.size} | Ndrc tax codes: ${ndrcDetails
-                              .map(s => s.taxType)
-                              .mkString("|")}"
-                          )
-                          Redirect(baseRoutes.IneligibleController.ineligible())
-                        }
-                    }
-                  case None                  => Redirect(routes.EnterMovementReferenceNumberController.enterMrn())
-                }
-            },
-          ifComplete =>
-            fillingOutClaim.draftClaim.movementReferenceNumber match {
-              case Some(value) =>
-                value match {
-                  case Left(_)  =>
-                    Ok(
-                      selectDutiesPage(
-                        SelectDutiesController
-                          .selectDutiesForm(ifComplete.dutiesSelected)
-                          .fill(ifComplete.dutiesSelected),
-                        dutiesForEntryNumber
-                      )
-                    )
-                  case Right(_) =>
-                    Ok(
-                      selectDutiesPage(
-                        SelectDutiesController
-                          .selectDutiesForm(ifComplete.dutiesSelected)
-                          .fill(ifComplete.dutiesSelected),
-                        duties(fillingOutClaim)
-                      )
-                    )
-                }
-              case None        =>
-                logger.warn("Could not find movement or entry reference number")
-                errorHandler.errorResult()
-            }
+      withAnswers[DutiesSelectedAnswer] { (fillingOutClaim, previousAnswer) =>
+        getAvailableDuties(fillingOutClaim).fold(
+          error => {
+            logger.warn("No Available duties: ", error)
+            Redirect(baseRoutes.IneligibleController.ineligible())
+          },
+          dutiesAvailable => {
+            val emptyForm  = selectDutiesForm(DutiesSelectedAnswer(dutiesAvailable))
+            val filledForm = previousAnswer.fold(emptyForm)(emptyForm.fill(_))
```

Comment:
i'd like to discuss how this guarantees that the form is filled correctly. let's have a call.

Reply:
Done